modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/CoreComponentBuildingDefinitionProvider.java

```diff
@@ -484,6 +484,7 @@ public void init() {
         .withSetterParameterDefinition(INHERIT_ITERABLE_REPEATABILITY,
                                        fromSimpleParameter(INHERIT_ITERABLE_REPEATABILITY).build())
         .withRegistrationName(OBJECT_MULE_CONFIGURATION)
+        .alwaysEnabled(true)
```

Comment:
If we do this we can get rid of the `resetMuleConfiguration();` in LazyMuleArtifactContext as it will be always disposed/unregistered before it gets registered again.